src/libsystemd/sd-netlink/netlink-message.c

```diff
@@ -520,7 +519,7 @@ int sd_netlink_message_open_container(sd_netlink_message *m, unsigned short type
 
         assert_return(m, -EINVAL);
         assert_return(!m->sealed, -EPERM);
-        assert_return(m->n_containers < RTNL_CONTAINER_DEPTH, -ERANGE);
+        assert_return(m->n_containers < (RTNL_CONTAINER_DEPTH - 1), -ERANGE);
```

Comment:
hmm, can you explain?

Reply:
`m->containers[m->n_containers + 1]` is accessed both in read and write. If `m->n_containers` is equal to `(RTNL_CONTAINER_DEPTH - 1)` the array is accessed out of bound

Reply:
please add this as comment here. At least on the first occasion this appears.